xarray/tests/test_datatree.py

```diff
@@ -1278,22 +1279,23 @@ def test_inherited_coords_index(self):
         assert "x" in dt["/b"].coords
         xr.testing.assert_identical(dt["/x"], dt["/b/x"])
 
-    def test_inherited_coords_override(self):
+    def test_inherit_only_index_coords(self):
         dt = DataTree.from_dict(
             {
-                "/": xr.Dataset(coords={"x": 1, "y": 2}),
-                "/b": xr.Dataset(coords={"x": 4, "z": 3}),
+                "/": xr.Dataset(coords={"x": [1], "y": 2}),
+                "/b": xr.Dataset(coords={"z": 3}),
```

Comment:
As we're planning to eventually change the API of the `Dataset` constructor to make it more explicit which coordinates have indexes, it might be nice to at least add comments here to describe the intention.

(sort of related to https://github.com/pydata/xarray/issues/8959)